deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -279,21 +373,15 @@ def _compact_tables(
         table = table.sort_by(input.sort_keys)
 
     compacted_table = None
-
-    if (
-        input.round_completion_info
-        and input.round_completion_info.hb_index_to_entry_range
-        and input.round_completion_info.hb_index_to_entry_range.get(str(hb_idx))
-        is not None
-    ):
+    if _does_hash_bucket_idx_have_compacted_table(input, hb_idx):
         compacted_table = _download_compacted_table(
```

Comment:
This `_does_hash_bucket_idx_have_compacted_table` is called twice if it returns True, then we download the table twice?